### ISSUE TITLE
Out of memory API errors after upgrading to 1.0.0-beta4

### ISSUE THREAD

**Issue by TomaszG** [sorry-cypress/sorry-cypress#170]
With a relatively huge database (2.482GB, 2797 runs recorded) the API throws the following errors (which results in a crash) when browsing the dashboard after upgrading to 1.0.0-beta4:

```
10/12/2020 2:33:38 PM
10/12/2020 2:33:38 PM<--- Last few GCs --->
10/12/2020 2:33:38 PMll[21:0x5631311c1cc0]   107140 ms: Mark-sweep 2049.2 (2052.9) -> 2048.4 (2053.6) MB, 1400.5 / 0.0 ms  (+ 143.8 ms in 27 steps since start of marking, biggest step 11.3 ms, walltime since start of marking 1569 ms) (average mu = 0.087, current mu = 0.016) all[21:0x5631311c1cc0]   108768 ms: Mark-sweep 2050.0 (2053.6) -> 2049.4 (2054.4) MB, 1480.4 / 0.0 ms  (+ 123.2 ms in 25 steps since start of marking, biggest step 11.8 ms, walltime since start of marking 1628 ms) (average mu = 0.049, current mu = 0.015) all
10/12/2020 2:33:38 PM
10/12/2020 2:33:38 PM<--- JS stacktrace --->
10/12/2020 2:33:38 PM
10/12/2020 2:33:38 PM==== JS stack trace =========================================
10/12/2020 2:33:38 PM
10/12/2020 2:33:38 PM    0: ExitFrame [pc: 0x56312df03fcd]
10/12/2020 2:33:38 PMSecurity context: 0x2522746808d1 <JSObject>
10/12/2020 2:33:38 PM    1: slice [0x1d3b566747d9] [buffer.js:~604] [pc=0x1370aeeca8a6](this=0x1d3b56673db9 <Object map = 0xf038666459>,0x01325337fc41 <Uint8Array map = 0xf0386659b9>,14029123,14029130)
10/12/2020 2:33:38 PM    2: deserializeObject(aka deserializeObject) [0x14f84b028129] [/app/node_modules/bson/lib/bson/parser/deserializer.js:~43] [pc=0x1370aed8d1af](this=0x3a68463804b1 <undefined>,0...
10/12/2020 2:33:38 PM
10/12/2020 2:33:38 PMFATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
```

This could be helpful for handling the issue manually: https://github.com/sorry-cypress/sorry-cypress/issues/60

In the meantime, I'm gonna clean-up old runs manually.

**Comment by agoldis** [comment]
Thanks for reporting that! 
First, I am happy to see that sorry-cypress was able to handle 2797 tests. 

Second, sure, will take a look!

**Comment by TomaszG** [comment]
Just an update - I reduced the size to 780MB, however, I'm still getting a few crashes each day. It seems that there's some kind of memory leak in API as its container memory usage goes up to 9GB before it crashes. Here are my current DB stats:
```
> db.runs.count()
242
> db.instances.count()
32823
> db.projects.count()
1
> db.stats()
{
        "db" : "sorry-cypress",
        "collections" : 3,
        "views" : 0,
        "objects" : 33072,
        "avgObjSize" : 18926.177249637156,
        "dataSize" : 625926534,
        "storageSize" : 284561408,
        "numExtents" : 0,
        "indexes" : 6,
        "indexSize" : 4714496,
        "fsUsedSize" : 1215061082112,
        "fsTotalSize" : 1947173228544,
        "ok" : 1
}
```

```
root@794cc28bcf04:/# du -sh /data/db
780M    /data/db
```

**Comment by agoldis** [comment]
Still didn't get to it... 

Running profiling on API's node process can help. If you can collect few samples - that would be helpful

**Comment by agoldis** [comment]
Not much luck after collecting few profiles / snapshotting API service - I haven't been able to see any obvious problem. My memory leak troubleshooting powers and not great to be honest. 

Probably it needs much bigger DB size or multiple clients connected... It would be great to collect a snapshot of API node process in the wild with a representative load. 

I am kind of clueless at this stage. I suspect that polling and Apollo Server cache are to blame, but yet to validate it.

@TomaszG do you, by any chance, can do that? Basically - it requires running the API with `--inspect`, connecting with Chrome Debugger and collecting snapshots / recordings.

**Comment by agoldis** [comment]
May be https://github.com/sorry-cypress/sorry-cypress/issues/202 has something to do with the memory issue

**Comment by agoldis** [comment]
@TomaszG Can you please try https://github.com/sorry-cypress/sorry-cypress/releases/tag/v1.0.0-beta.10

**Comment by agoldis** [comment]
@TomaszG any luck?

**Comment by agoldis** [comment]
Please check the latest version, it should improve memory situation. Feel free to reopen!

**Comment by TomaszG** [comment]
Sorry, I was not able to check it before. It seems that the issue has been fixed, since 1.0.0-beta.10 I don't see any crashes in service logs anymore, I will reopen the issue if there are any. Thanks!
